vector/src/main/java/im/vector/app/features/spaces/leave/SpaceLeaveAdvancedFragment.kt

```diff
@@ -44,24 +47,88 @@ class SpaceLeaveAdvancedFragment @Inject constructor(
+                    getString(R.string.space_leave_prompt_msg_only_you)
+                } else if (state.isLastAdmin) {
+                    getString(R.string.space_leave_prompt_msg_as_admin)
+                } else if (!summary.isPublic) {
+                    getString(R.string.space_leave_prompt_msg_private)
+                } else {
+                    null
+                }
+
+                views.spaceLeavePromptDescription.isVisible = warningMessage != null
+                views.spaceLeavePromptDescription.text = warningMessage
+            }
+
+            views.spaceLeavePromptTitle.text = getString(R.string.space_leave_prompt_msg_with_name, state.spaceSummary?.name ?: "")
+        }
+
         views.roomList.configureWith(controller)
         views.spaceLeaveCancel.debouncedClicks { requireActivity().finish() }
 
         views.spaceLeaveButton.debouncedClicks {
             viewModel.handle(SpaceLeaveAdvanceViewAction.DoLeave)
         }
 
-        views.publicRoomsFilter.queryTextChanges()
-                .debounce(100)
-                .onEach {
-                    viewModel.handle(SpaceLeaveAdvanceViewAction.UpdateFilter(it.toString()))
+        views.spaceLeaveSelectGroup.setOnCheckedChangeListener { _, optionId ->
+            run {
+                when (optionId) {
+                    R.id.spaceLeaveSelectAll  -> {
+                        viewModel.handle(SpaceLeaveAdvanceViewAction.SelectAll)
+                    }
+                    R.id.spaceLeaveSelectNone -> {
+                        viewModel.handle(SpaceLeaveAdvanceViewAction.SelectNone)
+                    }
                 }
-                .launchIn(viewLifecycleOwner.lifecycleScope)
+            }
+        }
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        menu.findItem(R.id.menu_space_leave_search)?.let { searchItem ->
+
+            searchItem.setOnActionExpandListener(object : MenuItem.OnActionExpandListener {
+                override fun onMenuItemActionExpand(item: MenuItem?): Boolean {
+                    viewModel.handle(SpaceLeaveAdvanceViewAction.SetFilteringEnabled(isEnabled = true))
+                    return true
+                }
+
+                override fun onMenuItemActionCollapse(item: MenuItem?): Boolean {
+                    viewModel.handle(SpaceLeaveAdvanceViewAction.SetFilteringEnabled(isEnabled = false))
+                    return true
+                }
+            })
+
+            val searchView = searchItem.actionView as SearchView
+            searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                override fun onQueryTextSubmit(query: String?): Boolean {
+                    return true
+                }
+
+                override fun onQueryTextChange(newText: String?): Boolean {
+                    viewModel.handle(SpaceLeaveAdvanceViewAction.UpdateFilter(newText ?: ""))
+                    return true
+                }
+            })
+        }
```

Comment:
I doesn't feel like this is necessary here, but this could be just because I'm used to read anonymous classes declaration :) It feels odd to create a builder (as it looks in your suggestion) and extension functions just to hide object declaration. @ouchadam i think we need a third voice here to decide

Reply:
I'm quite easy on this, from the syntax perspective it could be nice to create a declarative api on top of the `SearchItem` (similar to what [android-ktx](https://developer.android.com/kotlin/ktx/extensions-list) is doing and from a quick search it looks like we're using the `SearchItem` in a few places

a middle ground could be creating a local function to abstract away the boilerplate, with a potential future tech task to refactor the other usages and further improve the api 

```kotlin
private fun SearchItem.bind(onExpanded: () -> Unit, onCollapsed: () -> Unit, onTextChanged: (String) -> Unit) {
  setOnActionExpandListener(...)
  ...
}

searchItem.bind(
  onExpanded = { ... },
  onCollapsed = { ... },
  onTextChange = { ... } 
)
```